### ISSUE TITLE
🐛 BUG: duplicate styles in production mode for rehydrated components

### ISSUE THREAD

**Issue by demil** [snowpackjs/astro#519]
### Bug Report: Quick Checklist

- [:+1:] I am on the latest version of Astro & all plugins.
- [:+1:] I am on an ESM Node.js version (`v12.20.0+, v14.13.1+, or v16.0.0+`)
- [:+1:] I use package manager: (Fill in: npm, yarn, pnpm, etc).
- [Mac] I run Astro on OS: (Fill in: Windows, Mac, Linux, etc).

### Describe the bug

In production mode styles of components are duplicated in head.
In rendered page there is common.css where all components styles are already and in the component js there is some code that is similar to css.module.proxy.js that is adding those styles again in head in <style> tags

### Steps to reproduce

Run astro build with some components that are rehydrated and check out <head> and code in those components, you fill find duplicate styles in head and parts of css.module.proxy,js code.

### Expected behavior

Styles should not be duplicated

**Comment by matthewp** [comment]
I just tested this in a fresh project and didn't get duplicate styles. Can you try again and let us know @demil ?

**Comment by chrisbrown-io** [comment]
I'm having the same issue/can replicate using the svelte template.

```bash
mkdir astro-svelte-styles-test
cd astro-svelte-styles-test
npm init astro
# setup 'Starter Kit (Generic)' + 'Svelte'
npm i
```

Add any style to `src/components/SvelteCounter.svelte`, e.g:

```svelte
<script>
  let count = 0;

  function add() {
    count += 1;
  }

  function subtract() {
    count -= 1;
  }
</script>

<style>
  #svelte {
    background: red;
  }
</style>

<div id="svelte" class="counter">
  <button on:click={subtract}>-</button>
  <pre>{ count }</pre>
  <button on:click={add}>+</button>
</div>

```

When viewing the built site, the components style will appear in the extracted/concatenated stylesheet linked within the SSR'ed page (e.g `/_astro/index-Z1yeaK8.css`), along with the other component and page styles – which is as expected.

However, the style is then also included as a string within the components source - which when loaded for hydration, injects an inline `<style></style>` tag with the same styles as included in the concatenated stylesheet:

```javascript
if ("undefined" != typeof document) {
  const e = "#svelte.svelte-1ylhmyx{background:red}"
    , t = document.createElement("style")
    , s = document.createTextNode(e);
  t.type = "text/css",
  t.appendChild(s),
  document.head.appendChild(t)
}
```

I'm not super knowledgeable of the internals of snowpack, but I've tried to do a bit of digging to see if I could find a solution. When looking at the component source, it seems as though at some point during the build process, snowpack is adding a proxy import for the svelte component styles (e.g `import './SvelteCounter.svelte.css.proxy.js';`). This then gets transformed into the former 'injection' code, which ends up in the component source code – I presume as a result of trying to inline the import. It seems as though snowpacks [generateCssImportProxy](https://github.com/snowpackjs/snowpack/blob/main/snowpack/src/build/build-import-proxy.ts#L149) is responsible for generating this code. I assume that snowpack/astro is picking up svelte's statically generated CSS and assuming it hasn't been extracted seperately by the astro build process?

**Comment by jasikpark** [comment]
I can confirm that @chrisbrown-io's reproduction is accurate: https://codesandbox.io/s/confident-williams-c3tfz?file=/src/components/Counter.svelte

The svelte CSS is added via `<link rel="stylesheet" href="/_astro/src/components/Counter.svelte.css">` and via `<style type="text/css">#svelte.svelte-1ylhmyx{background:red}</style>` added to the end of `<head>` when the svelte component is hydrated - I doubled checked it was at hydrate by disabling js

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
